cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedTable.java

```diff
@@ -180,6 +174,35 @@ public Scanner readByIndex(byte[] column, byte[] value) {
   }
 
   /**
+   * Reads table rows within the given secondary index key range. If no rows are indexed, falling within the given
+   * range, then a {@link co.cask.cdap.api.dataset.table.Scanner} with no results will be returned.
+   *
+   * @param column the column to use for the index lookup
+   * @param startValue the inclusive start of the range for which rows must fall within to be returned in the scan.
+   *                   {@code null} means start from first row of the table
+   * @param endValue the exclusive end of the range for which rows must fall within to be returned in the scan
+   *                 {@code null} means end with the last row of the table
+   * @return a Scanner returning rows from the data table, whose stored value for the given column is within the the
+   * given range.
+   * @throws java.lang.IllegalArgumentException if the given column is not configured for indexing.
+   */
+  public Scanner scanByIndex(byte[] column, @Nullable byte[] startValue, @Nullable byte[] endValue) {
+    assertIndexedColumn(column);
+    // keyDelimiter is not used at the end of the rowKeys, because they are used for a range scan,
+    // instead of a fixed-match lookup
+    byte[] startRow = startValue == null ? null : Bytes.concat(column, keyDelimiter, startValue);
```

Comment:
When start value is null, we don't want to scan from the start of the table, we want to scan from the start of the entries for this column.  So start row should be column + keyDelimiter.

As long as we don't make keyDelimiter configurable, end row can be column + (keyDelimiter +1).  Theoretically we could use filters to do a prefix match on the row keys, but the Table API does not support filters.